openvpn-wire-protocol.xml

```diff
@@ -1542,6 +1544,226 @@ struct datakeys {
+      not only with the current MTU size and use L accordingly in the calcualations. E.g. the current OpenVPN implementation
+      will always try to send up to the internal buffer size of 1600 byte in the default configuration.
+    </t>
+
+    <t>
+      Example calculations for confidentialy limit. This is done do to give an example for an implementation that
+      assumes a fixed L and to give an estimate how many packets can be encrypted assuming assuming a fixed packet
+      size.
+
+      Using the limits in the documents and p=2^-57 and L=2**10.
+
+      <artwork>q &le; (p^(1/2) * 2^(129/2) - 1) / (L + 1)></artwork>
+
+      gives us basically the same limit that TLS imposes:
+      <artwork> q &le; 2^(24.9999) </artwork>
+
+
+      When using L = 2^7 (packet/buffer size of 1500 rounded up to the nearest power of 2 an)
+      we get:
+
+      <artwork> q &le; 2^(26.999) </artwork>
+      
+    </t>
+    <t>
+      For the integrity limit
+        <artwork>v &le; min(2^64, (p * 2^127) / (L + 1))</artwork>
+
+
+        with p=2^-57, and setting v=3^36, we can have L to be up to 2^34
+        (256 GB). So for packet sizes used in the OpenVPN protocol
+      we ignore this formula as we are always within the safe limits.
+    </t>
+    <t>
+      The following table gives an overview of the limits for number of packets when calculating them with above confidentially formula.
+    </t>
+    <texttable>
+      <ttcol>packet size</ttcol>
+      <ttcol>Maximum number of packets</ttcol>
+      <c> 64 </c>
+      <c> 2^33 </c>
+      
+      <c> 1500 </c>
+      <c> 2^29.4 </c>
+      
+      <c> 1600 </c>
+      <c> 2^29.3 </c>
+      
+      <c>2048</c>
+      <c>2^29</c>
+      
+      <c>4096</c>
+      <c>2^28</c>
+
+      <c>9000</c>
+      <c>2^26.8</c>
+
+      <c>16384</c>
+      <c>2^26</c>
+
+      <c>65336</c>
+      <c>2^24</c>
+    </texttable>
+
+    <t>
+      The receiver should count the number of packets that have been received but failed verification.
+      The number of packets that fail verification and we can tolerate is specified differently
+      for DTLS 1.3 and the data usage limit document. For OpenVPN we set the number to the limit of 
+      2^36 packets failed verifications, the same as (D)TLS 1.3, which is lower than the theoretical limit.
+      An implementation may choose a lower number.
+    </t>
+      </section>
+      <section title="Epoch key generation">
+        <t>      
+          The epoch key generation is inspired by the TLS key generation and
+          <xref target="RFC5869">HMAC-based Key Derivation Function (HKDF)</xref>
+          and the definition of HKDF-Expand-Label in <xref target="RFC8446">TLS 1.3</xref>.
+        </t>
+
+        <t>
+          OpenVPN uses different epoch keys for the client to server and the server to client direction.
+          The epoch 1 key E_1 are the first 32 bytes the OpenVPN key key_c2s and key_s2c respectiviely.
```

Comment:
I don't understand it. `The epoch 1 key E_1 are the first 32 bytes` and `the OpenVPN key key_c2s and key_s2c respectiviely`. Is `of` missing in between? Also a typo in `respectiviely`.